src/app/analyze/stack/stack-details/stack-details.component.ts

```diff
@@ -234,66 +235,107 @@ export class StackDetailsComponent implements OnInit {
-        this.getComponents(this.stackAnalysesData[0].components);
-
-        this.packageName = this.stackAnalysesData[0].name;
-        this.packageVersion = this.stackAnalysesData[0].version;
-        this.averageUsage = this.stackAnalysesData[0].usage.average_usage;
-        this.lowPublicUsageComponents = this.stackAnalysesData[0].usage.low_public_usage_components;
-        this.redhatDistributedComponents = this.stackAnalysesData[0].usage.redhat_distributed_components;
-
-        this.averageStars = this.stackAnalysesData[0].popularity.average_stars;
-        this.averageForks = this.stackAnalysesData[0].popularity.average_forks;
-        this.lowPopularityComponents = this.stackAnalysesData[0].popularity.low_popularity_components;
-
-        this.distinctLicenses = this.stackAnalysesData[0].distinct_licenses;
-        this.totalLicenses = this.stackAnalysesData[0].total_licenses;
-
-        this.totalSecurityIssues = this.stackAnalysesData[0].total_security_issues;
-        this.cvss = this.stackAnalysesData[0].cvss;
-
-        this.componentsWithTests = this.stackAnalysesData[0].metadata.components_with_tests;
-        this.componentsWithDependencyLockFile = this.stackAnalysesData[0].metadata.components_with_dependency_lock_file;
-        this.requiredEngines = this.stackAnalysesData[0].metadata.required_engines;
-        for (let key in this.requiredEngines) {
-          if (this.requiredEngines.hasOwnProperty(key)) {
-            this.requiredEnginesArr.push({ key: key, value: this.requiredEngines[key] });
-          }
-        }
-
-        for (let i = 0; i < this.stackAnalysesData[0].components.length; i++) {
-          let myObj: any = {};
-          myObj.ecosystem = this.stackAnalysesData[0].components[i].ecosystem;
-          myObj.pkg = this.stackAnalysesData[0].components[i].name;
-          myObj.version = this.stackAnalysesData[0].components[i].version;
-          myObj.latestVersion = this.stackAnalysesData[0].components[i].latest_version;
-          myObj.publicUsage = this.stackAnalysesData[0].components[i].dependents_count;
-          myObj.relativePublicUsage = this.stackAnalysesData[0].components[i].relative_usage;
-          if (this.stackAnalysesData[0].components[i].github_details.forks_count) {
-            myObj.popularity = this.stackAnalysesData[0].components[i].github_details.forks_count + "/" + this.stackAnalysesData[0].components[i].github_details.stargazers_count;
-          } else {
-            myObj.popularity = '';
-          }
+      myObj.redhatUsage = '';
+      myObj.licence = components[i].licenses[0];
+      this.componentsDataTable.push(myObj);
+    }
+  }
 
-          myObj.redhatUsage = '';
-          myObj.licence = this.stackAnalysesData[0].components[i].licenses[0];
-          this.componentsDataTable.push(myObj);
-        }
+  private setStackMetrics(stackData: any): void {
```

Comment:
Instead of individual assignments, can we have something like selectedStackData and use selectedStackData.name, selectedStackdata.version, etc in the html?